pkg/manifests/config.go

```diff
@@ -467,6 +476,35 @@ func (c *Config) NoProxy() string {
 	return c.ClusterMonitoringConfiguration.HTTPConfig.NoProxy
 }
 
+func (c *Config) LoadEnforcedBodySizeLimit(pcr PodsCapacityReader, ctx context.Context) error {
+	enforcedBodySizeLimit := c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit
+	if enforcedBodySizeLimit == "" {
+		podCapacity, err := pcr.PodsCapacity(ctx)
+		if err != nil {
+			return fmt.Errorf("error fetching pod capacity: %v", err)
+		}
+		enforcedBodySizeLimit = calculateBodySizeLimit(podCapacity)
+	}
+	if err := pm.ValidateSizeField(enforcedBodySizeLimit); err != nil {
+		return err
+	}
+	c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit = enforcedBodySizeLimit
+	return nil
+}
+
+func calculateBodySizeLimit(podCapacity int) string {
+	const samplePerPod = 50
+	const sizePerSample = 200 // 200 Bytes
```

Comment:
These consts don't match with the comment from `minimalSizeLimit`. We need to use the same inputs and probably document in more details how we got these numbers. 

```
// Limit the body size from scrape queries
// Assumptions: one node has maximum 110 pods, each pod exposes 100 metrics, each metric is expressed by at most 100 bytes.
// 1.5x the size for a safe margin, it rounds up to 2MB.
```

Reply:
Going to check again on average size of metrics and count of metrics per pod. Then update the comments.